templates/targets/x86_64/generic/disk-config.nix

```diff
@@ -1,6 +1,7 @@
-# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
 # SPDX-License-Identifier: Apache-2.0
-# Example to create a bios compatible gpt partition
+#
+# The partitioning scheme is hardcoded for 500G disk.
```

Comment:
With this comment we could actually make this specific to Lenovo X1 Carbon Gen11 SKU (stock keeping unit) with 500G of internal NVMe. I'm afraid we must build target specific installers as we have not yet described disk space extension in case the internal NVMe is larger. Optionally, the installer could resolve the size of the disk with `lsblk` and parametrize the size to generate disk config for `generic` use. I think all the other fixed sizes could remain the same and only the remaining space here https://github.com/tiiuae/ghaf/pull/478/files#diff-8a1fbff8658406e95003d28116ba21ced73c81b4fb8d1a5edad5fe5aeef1c427R154 would have to be calculated. Please check with  @remimimimimi if this is feasible. The next iterations of installer must support other installer parameters - like user specific secrets - as well.

Reply:
> Optionally, the installer could resolve the size of the disk with lsblk and parametrize the size to generate disk config for `generic` use.

This exact approach assumes that the installer program changes `disk-config.nix` file before passing it into `disko` utility, which I am pretty sure we want to avoid.

> I think all the other fixed sizes could remain the same and only the remaining space <here> would have to be calculated.

Unfortunately, `disko` utility for some reason does not support `size = "100%"` parameter in that exact place. It fails with error while partitioning the disk. Probably I should have mentioned this in the comment inside the file itself.

> The next iterations of installer must support other installer parameters - like user specific secrets - as well.

This is out of scope of this PR.

Reply:
> Unfortunately, disko utility for some reason does not support size = "100%" parameter in that exact place. It fails with error while partitioning the disk. Probably I should have mentioned this in the comment inside the file itself.

It seems that [this should work](https://github.com/nix-community/disko/commit/e1d5b132d1ac75b7f87aa407024fdb24766e695f) @unbel13ver .

Reply:
Regarding secrets one can use the `diskoImagesScript` to generate full systems that also contain secrets. The script is run outside of the NixOS sandbox.

Reply:
> My proposal for placement is <ghaf_repo_root>/targets/lenovo-x1-carbon-disk-config.nix because this is very specific to that device.

That is reasonable, implemented.

Also I managed to take the rest of the disk in use for StorageVM (should have added "100%FREE" instead of "100%").
@leivos-unikie could you retest this please? I've updated the expected partitioning in the header of this PR.

Reply:
I spotted these deviations in log messages

- at booting installer
![failed_creating_link](https://github.com/tiiuae/ghaf/assets/129752236/fe7960b0-84d1-4f5e-b68b-ab9ff19f8de7)

- in the beginning of installation process
![dont_know_msg](https://github.com/tiiuae/ghaf/assets/129752236/5625136a-fb97-4185-b759-e8bb0961b2b2)

- after installation was finished and pressed Enter for reboot
![failed_unmounting](https://github.com/tiiuae/ghaf/assets/129752236/c1d20ba7-0ba8-441e-8d3b-63cf53b9c870)


Otherwise no issues found in testing.

Reply:
Thanks for testing! Those are not related to the matter of this PR (those are not something worth to worry about, anyway).